### ISSUE TITLE
Add "Where to go from here?" to cog creation guide

### ISSUE THREAD

**Issue by Jackenmen** [Cog-Creators/Red-DiscordBot#6134]
### What component of Red (cog, command, API) would you like to see improvements on?

"Creating cogs for Red V3" document

### Describe the enhancement you're suggesting.

The current document doesn't really go beyond creating a simple "hello world" cog. We mention the Red V2 migration guide in the "Additional resources" section but that probably isn't useful to the vast majority of people reading this document these days.

I think we should replace that with a "Where to go from here?" section (could also just keep the more boring "Additional resources" name :P) describing how you can take the next step by mentioning discord.py documentation, a few core APIs (especially the documents about Config, Data Manager, Utility Functions, commands package, and the bot class), and mentioning that more documentation can be found in the sidebar. Maybe some general pointers on how to navigate d.py's and Red's documentation could be helpful as well, unsure.
This section should probably go *above* the "Publish your cog" and "Becoming an Approved Cog Creator" sections.

**Comment by Frost-4X** [comment]
Perhaps an advanced walkthrough of creating a complex cog, I've found that learning by doing is the best for a lot of people, myself included. It would only need to cover core concepts.

**Comment by Jackenmen** [comment]
> Perhaps an advanced walkthrough of creating a complex cog

Yeah, this *could* be worth doing though it might also be worth having a bit more detailed tutorials like the one we have in Config's documentation for other bot components. This also makes me think that a lot of people learn by example and so we should consider mentioning that you could learn by looking at the source code of other people's approved cogs. This does come with a caveat that not all such cogs have the best code quality *but* approved cogs do have to follow common cog practices so that's good.
